osquery/tables/system/windows/registry.cpp

```diff
@@ -72,6 +72,21 @@ void explodeRegistryPath(const std::string& path,
   rKey = osquery::join(toks, kRegSep);
 }
 
+Status sanitizeRegistryStrings(const DWORD dataType,
+                               BYTE* dataBuff,
+                               DWORD dataSize) {
+  if (!dataBuff ^ (dataSize == 0)) {
```

Comment:
I don't think this will do what you're wanting, as I'm pretty sure you can't use logical XOR's in conditional statements. Further, I'm curious if this is needed? It's looking like the condition you're wanting is that `dataBuff` is not a nullpointer, _and_ that `dataSize > 0`, so wouldn't the next conditional verify that? I think the first reason I commented on this was I was curious if there'd ever be a circumstance where `dataBuff` wouldn't be a nullptr, and `dataSize` would be zero or vice versa? in that case couldn't `dataBuff == nullptr || dataSize == 0` suffice? I feel like I'm missing something, but overall couldn't we remove this conditional check and rely on the one following? And possibly throw the `return Status(1, ...` into an `else`?